src/kirby/components/shared/resize-observer/resize-observer.service.ts

```diff
@@ -0,0 +1,67 @@
+import { ElementRef, Injectable, OnDestroy, NgZone } from '@angular/core';
+
+import { ResizeObserver } from './types/resize-observer';
+import { ResizeObserverEntry } from './types/resize-observer-entry';
+import { ResizeObserverFactory } from './resize-observer.factory';
+
+@Injectable({ providedIn: 'root' })
+export class ResizeObserverService implements OnDestroy {
+  // IMPORTANT: Keeps track of the existing ResizeObserver so it can be reused for performance.
+  // See: https://groups.google.com/a/chromium.org/forum/#!msg/blink-dev/z6ienONUb5A/F5-VcUZtBAAJ
+  private observer: ResizeObserver | null;
+  private observedElements = new WeakMap<Element, (entry: ResizeObserverEntry) => void>();
+
+  constructor(private _resizeObserverFactory: ResizeObserverFactory, private zone: NgZone) {}
+
+  observe(elementOrRef: Element | ElementRef<Element>, action: (entry: ResizeObserverEntry) => void): Element {
+    if (this.observer === undefined) {
+      this.observer = this._resizeObserverFactory.create((entries) => this.handleResize(entries));
+    }
+
+    const element = elementOrRef instanceof ElementRef ? elementOrRef.nativeElement : elementOrRef;
+    if (!this.observedElements.has(element)) {
+      if (this.observer) {
+        // console.log(`Observing <${element.tagName}> through ResizeObserver...`);
+        // IMPORTANT: Has to be run outside the Angular zone, for it to work with ResizeObserver polyfill:
+        this.zone.runOutsideAngular(() => {
+          this.observer.observe(element);
+        });
+      }
+      this.observedElements.set(element, action);
+    }
+    return element;
+  }
+
+  private handleResize(entries: ResizeObserverEntry[]) {
+    entries.forEach(entry => {
+      const action = this.observedElements.get(entry.target);
+      if (action) {
+        action(entry);
+      }
+    });
+  }
+
+  ngOnDestroy() {
+    if (this.observer) {
+        // IMPORTANT: Has to be run outside the Angular zone, for it to work with ResizeObserver polyfill:
+        this.zone.runOutsideAngular(() => {
```

Comment:
The indentation is a bit off here.

Reply:
Not understood?

Reply:
I am also unsure what i am talking about here. So just ignore.